tools/generate-bolts.py

```diff
@@ -197,9 +200,10 @@ def __init__(self, name):
         self.depends_on = {}
         # FIXME: internal msgs can be enums
         self.is_enum = False
+        self.type_comments = []
 
-    def add_data_field(self, field_name, type_obj, count=1, is_optional=[]):
-        FieldSet.add_data_field(self, field_name, type_obj, count, is_optional)
+    def add_data_field(self, field_name, type_obj, count=2, is_optional=[], comments=[]):
```

Comment:
Just for my own understanding: what is `count` used for, and why is it being bumped here?

Reply:
This is a mistake, it should be 1. It's used to mark whether or not a field is variable-length.

Reply:
> I got lost a couple of times trying to reverse engineer the code (python code in particular and some of the accessors whose use I couldn't understand). It'd be nice to have more comments explaining the rationale behind the individual parts

Agreed that more comments is probably useful. I'd also recommend checking out the generated test case code; it's fairly well documented wrt the scope of what all the cases that the generators cover. 

    make check-tools

will generate four files (and sort of check that they compile): `tools/test/gen_wire.[c,h]` and `tools/test/gen_print.[c,h]`